resources/assets/js/components/Application/BasicInfo/BasicInfoPage.tsx

```diff
@@ -1,59 +1,121 @@
 import { getLocale } from "../../../helpers/localize";
 import { navigate } from "../../../helpers/router";
 import {
   applicationIndex,
   applicationExperienceIntro,
   applicationWelcome,
 } from "../../../helpers/routes";
-import makeProgressBarSteps from "../ProgressBar/progressHelpers";
-import ProgressBar, { stepNames } from "../ProgressBar/ProgressBar";
-import { fakeApplication } from "../../../fakeData/fakeApplications";
-import BasicInfo from "./BasicInfo";
-import fakeJob from "../../../fakeData/fakeJob";
+import { Job, ApplicationNormalized } from "../../../models/types";
+import { DispatchType } from "../../../configureStore";
+import { RootState } from "../../../store/store";
+import {
+  getApplication,
+  getApplicationIsUpdating,
+} from "../../../store/Application/applicationSelector";
+import {
+  updateApplication as updateApplicationAction,
+  fetchApplicationNormalized,
+} from "../../../store/Application/applicationActions";
+import { fetchJob } from "../../../store/Job/jobActions";
+import { getJob, getJobIsUpdating } from "../../../store/Job/jobSelector";
 
 interface BasicInfoPageProps {
   applicationId: number;
 }
 
-export const BasicInfoPage: React.FunctionComponent<BasicInfoPageProps> = ({
+const BasicInfoPage: React.FunctionComponent<BasicInfoPageProps> = ({
   applicationId,
 }) => {
   const intl = useIntl();
   const locale = getLocale(intl.locale);
+  const dispatch = useDispatch<DispatchType>();
 
-  const application = fakeApplication(); // TODO: get real application.
-  const job = fakeJob(); // TODO: Get real job associated with application.
+  // Load Application.
+  const applicationSelector = (
+    state: RootState,
+  ): ApplicationNormalized | null => getApplication(state, { applicationId });
+  const application = useSelector(applicationSelector);
+  const applicationIsUpdating = useSelector((state: RootState) =>
+    getApplicationIsUpdating(state, { applicationId }),
+  );
 
-  const handleContinue = (values): void => {
-    // TODO: Save basic info form values.
+  useEffect(() => {
+    if (application === null && !applicationIsUpdating) {
+      dispatch(fetchApplicationNormalized(applicationId));
+    }
+  }, [dispatch, applicationId, application, applicationIsUpdating]);
+
+  // Load Job.
+  const jobId = application?.job_poster_id;
+  const jobSelector = (state: RootState): Job | null =>
+    jobId ? getJob(state, { jobId }) : null;
+  const job = useSelector(jobSelector);
+  const jobUpdatingSelector = (state: RootState): boolean =>
+    jobId ? getJobIsUpdating(state, jobId) : false;
+  const jobIsUpdating = useSelector(jobUpdatingSelector);
+
+  useEffect(() => {
+    // If job is null and not already updating, fetch it.
+    if (jobId && job === null && !jobIsUpdating) {
+      dispatch(fetchJob(jobId));
+    }
+  }, [jobId, job, jobIsUpdating, dispatch]);
+
+  if (application === null || job === null) {
+    return null;
+  }
```

Comment:
Is returning null really how we want to handle this? We could render the progress bar before these are loaded, and show some kind of loading state...

Reply:
Ideally yeah, we should get some loading UI from design